twisted/names/test/test_rootresolve.py

```diff
@@ -209,20 +209,20 @@ def test_lookupChecksClass(self):
         badClass.CLASS = HS
         servers = {
             ('1.1.2.3', 53): {
-                ('foo.example.com', A): {
-                    'answers': [('foo.example.com', badClass)],
-                    'authority': [('foo.example.com', Record_NS('ns1.example.com'))],
-                    'additional': [('ns1.example.com', Record_A('10.0.0.2'))],
+                (b'foo.example.com', A): {
+                    'answers': [(b'foo.example.com', badClass)],
+                    'authority': [(b'foo.example.com', Record_NS(b'ns1.example.com'))],
+                    'additional': [(b'ns1.example.com', Record_A('10.0.0.2'))],
```

Comment:
Sure, but that's not really relevant to the review. What I'm interested in is not just hitting this patch with a hammer until the tests pass and calling it fine: I want us to step back and think about the way `names` handles strings and string types to make sure it makes some degree of sense.